nodes/src/nodes/approval/notifier.py

```
@@ -0,0 +1,185 @@
+# ApprovalNotifier -- sends notifications when an approval is requested.
+#
+# Supports three channels: 'webhook', 'log', and 'none'.
+# The webhook channel prepares a JSON payload and POSTs it to the configured
+# URL.  The HTTP call is fire-and-forget with a short timeout so it does not
+# block the pipeline hot path.  Delivery failures are logged but never raise.
+# ------------------------------------------------------------------------------
+
+import json
+import logging
+import urllib.request
+from typing import Any, Dict, Optional
```

Ruff (0.15.7)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)